test/unit/webrtc.test.ts

```diff
@@ -1951,7 +1951,52 @@ describe('handleGenesysOffer', () => {
     expect(spy).not.toHaveBeenCalled();
     expect.assertions(3);
   });
-  
+
+  it('should NOT emit a new session for a reinvite if it is a duplicate offer.', async () => {
+    const iq: IQ = {
+      type: 'set',
+      from: '+155555555@org.gjoll.test',
+      genesysWebrtc: {
+        jsonrpc: '2.0',
+        method: 'offer',
+        id: '123testid',
+        params: {
+          sessionId: 'session24',
+          conversationId: 'cid',
+          sdp: 'a=ice-ufrag:asdfasdf\r\na=ice-pwd:fdsafdsa\r\n',
+          reinvite: true
+        }
+      }
+    };
+    webrtc.on('incomingRtcSession', (session: GenesysCloudMediaSession) => {
+      expect(session.setRemoteDescription).toHaveBeenCalled();
```

Comment:
Checking my understanding of the execution flow: do these have to be verified here or could they also be verified after `handleGenesysOffer` is called?